saleor/graphql/webhook/mutations.py

```diff
@@ -255,14 +256,22 @@ def perform_mutation(cls, _root, info, **data):
                     code=WebhookErrorCode.INVALID,
                 )
             try:
-                app.webhooks.get(id=object_id)
+                webhook = app.webhooks.get(id=object_id)
+                webhook.is_active = False
+                webhook.save(update_fields=["is_active"])
             except models.Webhook.DoesNotExist:
                 raise ValidationError(
                     f"Couldn't resolve to a node: {node_id}",
                     code=WebhookErrorCode.GRAPHQL_ERROR,
                 )
-
-        return super().perform_mutation(_root, info, **data)
+        try:
+            response = super().perform_mutation(_root, info, **data)
+        except IntegrityError:
```

Comment:
You are only deactivating the webhook if the delete mutation comes from an app, not the user